src/ra_fwd.h

```diff
@@ -48,7 +48,16 @@ using HANDLE = void*;
 
 namespace ra {
 
-using tstring = std::basic_string<TCHAR>;
+#if _MBCS
+using NativeStrType = std::string;
+#elif _UNICODE
+using NativeStrType = std::wstring;
+#else
+#error Unknown character set detected! Only MultiByte and Unicode are supported!
+#endif // _MBCS
+
+using Native_CStrType = NativeStrType::const_pointer;
+using tstring         = NativeStrType;
```

Comment:
Moved these here because they're `usings` (aliases). `usings `have the same semantics as `auto`.